logic.py

```diff
@@ -81,9 +82,10 @@ def tell(self, sentence):
         self.clauses.extend(conjuncts(to_cnf(sentence)))
 
     def ask_generator(self, query):
-        "Yield the empty substitution if KB implies query; else nothing."
+        "Return the empty substitution {} if KB entails query; else return False."
         if tt_entails(Expr('&', *self.clauses), query):
-            yield {}
+            yield {}  # Why use yield when you are not returning a generator?
```

Comment:
ask_generator is expected to return a generator(an iterable is also fine) for other methods to use. In this case there is only one element returned by the generator function. I think the comments must be a bit more explicit about that.